lib/Index/Index.cpp

```diff
@@ -1691,6 +1708,25 @@ bool IndexSwiftASTWalker::initFuncDeclIndexSymbol(FuncDecl *D,
   return false;
 }
 
+bool IndexSwiftASTWalker::initLiteralInitRefIndexSymbol(ValueDecl *D,
```

Comment:
This is almost `initFuncRefIndexSymbol` - is `ide::isBeingCalled` false and thus causing it to return early?

Reply:
Yes, `isBeingCalled` is only checking for `(Self)ApplyExpr`. Should I update it to support `LiteralExpr`s with non-null `initializer`s?

Reply:
And if I do that, should I update `Expr->getReferencedDecl` to return the initializer for subclasses of `LiteralExpr`? Right now they're all using the `NO_REFERENCE` macro to return `null` unconditionally.

Reply:
That seems reasonable to me. That combined with `SemaAnnotator` walking this should mean that `IndexSwiftASTWalker` will work as is, I believe.

Reply:
Ok, I think I've got this rewritten the way you described now; PTAL!